Media: Make filename checks less strict in 'wp_image_src_get_dimensions'

joemcgill opened the pull request:
This modifies the check for full size files so that only the basename is compared with the image `src` to avoid misses whenever the `src` path has been modified.

Fixes: https://core.trac.wordpress.org/ticket/52417
Props ianmjones

joemcgill committed:
Media: Make filename checks less strict in 'wp_image_src_get_dimensions'

This modifies the check for full size files so that only the basename
is compared with the image `src` to avoid misses whenever the `src` path
has been modified.

mukeshpanchal27 commented:
Hello,

As per the PHP Coding Standards document - https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions can we use **Yoda Conditions** here? 

`if ( strpos( $image_src, wp_basename( $image_meta['file'] ) ) !== false ) {`

Replace to

`if ( false !=== strpos( $image_src, wp_basename( $image_meta['file'] ) ) ) {`

hellofromtonya commented:
>As per the PHP Coding Standards document - https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions can we use Yoda Conditions here?

Yoda Conditions doesn't apply in this case. Why? There's no variable on either side of the conditional expression. Instead, the expression compares a boolean literal `false` to a final result of the functions. 

IMO the code is well-formed and understandable.

joemcgill committed:
Merge branch 'master' into 52417-make-full-size-dimension-checks-less-strict

joemcgill committed:
Media: Make filename checks less strict in 'wp_image_src_get_dimensions'

This modifies the check for full size files so that only the basename
is compared with the image `src` to avoid misses whenever the `src` path
has been modified.

hellofromtonya commented on src/wp-includes/media.php:
```diff
@@ -1604,7 +1604,7 @@ function wp_image_src_get_dimensions( $image_src, $image_meta, $attachment_id =
 	if (
 		isset( $image_meta['file'] ) &&
-		strpos( $image_src, $image_meta['file'] ) !== false
+		strpos( $image_src, wp_basename( $image_meta['file'] ) ) !== false
```
Looks good @joemcgill 👍 👏

joemcgill commented:
Fixed in https://core.trac.wordpress.org/changeset/50144
